crates/metadata/src/specs.rs

```diff
@@ -366,21 +393,25 @@ impl<S, P> ConstructorSpecBuilder<S, P> {
     /// Sets the documentation of the message specification.
     pub fn docs<D>(self, docs: D) -> Self
     where
-        D: IntoIterator<Item = &'static str>,
+        D: IntoIterator<Item = <F as Form>::String>,
+        F::String: DocString,
```

Comment:
```suggestion
        F::String: AsRef<[u8]>,
```
This might work to avoid requiring the `DocString` trait and the new impls.

Reply:
Neat idea!